test/custom_datatable_types_test.clj

```diff
@@ -0,0 +1,123 @@
+(ns custom-datatable-types-test
+  (:require [burpless :refer [datatable-type parameter-type run-cucumber step]]
+            [clojure.test :refer [deftest is]])
+  (:import (clojure.lang Keyword)
+           (io.cucumber.datatable DataTable)
+           (io.cucumber.docstring DocString)
+           (java.util List)))
+
+
+;; FROM TABLE
+(defrecord TicTacToeBoard [topLeft topMid topRight
+                           midLeft midMid midRight
+                           botLeft botMid botRight])
+
+
+(def from-table-glues
+  [(step :Given "I have the following tic-tac-toe board:"
+         ^{:datatable TicTacToeBoard}
+         (fn [state ^TicTacToeBoard board]
+           (assoc state :board board)))
+
+   (step :Then "the {keyword} square should be empty"
+         (fn [{:keys [board] :as state} ^Keyword board-cell]
+           (assert (= nil (board-cell board)))
+           state))
+
+   (step :Then "the {keyword} square should be {keyword}"
+         (fn the_square_should_be [{:keys [board] :as state} ^Keyword board-cell ^Keyword expected-player]
+           (let [actual-player (board-cell board)]
+             (assert (= expected-player actual-player)))
+           state))
+
+   (datatable-type {:to-type   TicTacToeBoard
+                    :from-type :table
+                    :transform (fn [^DataTable table]
+                                 (apply ->TicTacToeBoard (map keyword (.values table))))})])
```

Comment:
this is awesome. very easy to instantiate a `defrecord`. we can also add this bit to the readme

Reply:
Thanks! Awesome enough for an approval, perhaps? 😉